web/packages/core/src/i18n.ts

```diff
@@ -0,0 +1,123 @@
+import { FluentBundle, FluentResource } from "@fluent/bundle";
+import { negotiateLanguages } from "@fluent/langneg";
+import type { FluentVariable } from "@fluent/bundle";
+
+interface FileBundle {
+    [filename: string]: string;
+}
+
+interface LocaleBundle {
+    [locale: string]: FileBundle;
+}
+
+// This is automatically populated by `tools/bundle_texts.js` via a postbuild script
+const BUNDLED_TEXTS: LocaleBundle = {
+    /* %BUNDLED_TEXTS% */
+};
+
+const bundles: Record<string, FluentBundle> = {};
+
+for (const locale in BUNDLED_TEXTS) {
+    const bundle = new FluentBundle(locale);
+    const files = BUNDLED_TEXTS[locale];
+    if (files) {
+        for (const filename in files) {
+            const text = files[filename];
+            if (text) {
+                for (const error in bundle.addResource(
+                    new FluentResource(text)
+                )) {
+                    console.error(
+                        `Error in text for ${locale} ${filename}: ${error}`
```

Comment:
Nit: I do not understand the code and I do not know Fluent, but might it make sense to change it to this (adding `text` to the printed string):

`Error in text for ${locale} ${filename} ${text}: ${error}`

?

Reply:
It'll be a huge text file, that's a lot of spam. We can look up the text file ourselves.